### ISSUE TITLE
JNA issue in M1 architecture

### ISSUE THREAD

**Issue by szaffarano** [revolut-engineering/jooq-plugin#23]
It seems that due to [this issue](https://github.com/oshi/oshi/issues/1462) M1 architectures are not able to use the plugin, getting the following error

```
Caused by: java.lang.NoClassDefFoundError: Could not initialize class com.github.dockerjava.transport.DomainSocket                                         
        at com.github.dockerjava.okhttp.UnixSocketFactory.createSocket(UnixSocketFactory.java:21)                                                          
        at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.java:257)                                                               
        at okhttp3.internal.connection.RealConnection.connect(RealConnection.java:183)                                                                     
        at okhttp3.internal.connection.ExchangeFinder.findConnection(ExchangeFinder.java:224)                                                              
        at okhttp3.internal.connection.ExchangeFinder.findHealthyConnection(ExchangeFinder.java:108)                                                       
        at okhttp3.internal.connection.ExchangeFinder.find(ExchangeFinder.java:88)                                                                         
        at okhttp3.internal.connection.Transmitter.newExchange(Transmitter.java:169)                                                                       
        at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:41)                                                            
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)                                                               
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117)                                                               
        at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:94)                 
```

Bumping `docker-java-transport-okhttp` fixes the issue

```patch
-    implementation("com.github.docker-java:docker-java-transport-okhttp:3.2.8")
+    implementation("com.github.docker-java:docker-java-transport-okhttp:3.2.14")
```

As a workaround, it's possible to "override" this dependency (as well as commons-lang)

```kotlin
buildscript {
    configurations.classpath {
        resolutionStrategy {
            setForcedModules("org.jooq:jooq-codegen:3.14.3")
        }
    }

    dependencies {
       classpath("com.github.docker-java:docker-java-transport-okhttp:3.2.14")
       classpath("commons-lang:commons-lang:2.6")
    }
}
```

**Comment by szaffarano** [comment]
Hi @adrianskrobaczrevolut do you have any plans to make a new release including this change in the near future?
Thanks

**Comment by adrianskrobaczrevolut** [comment]
Hey @szaffarano 
Sorry for slight delay. We've been doing some internal testing along with other docker stability improvements. The fix has been released in version [0.3.9](https://github.com/revolut-engineering/jooq-plugin/releases/tag/0.3.9). One additional thing we've noticed is that the problem occurs when the `generateJooqClasses` is run from InteliJ(at least on MacOs). The root cause seems to be the fact the `jna` library used by the `docker-java` is trying to find native libraries in classpath and it works well from command line, but IntelliJ changes the classpath slightly and it makes library fail to locate the native library. It might be needed to stop the Gradle Daemons as well if the library failed to load the native libs once it will remain failed in the daemon lifetime. Another way to fix it permanently is to upgrade your JDK to 16 or above as it introduces `java.net.UnixDomainSocketAddress` class which `docker-java` takes advantage of.
